tui/src/ui/components/dlna_server.rs

```diff
@@ -134,4 +140,152 @@ impl Component<Msg, UserEvent> for DlnaServer {
+        };
+        node
+    }
+
+    fn mediaserver_item(id: String, value: String) -> RecVec<String, String> {
+        let mut node = RecVec {
+            id,
+            value,
+            children: Vec::new(),
+        };
+        node
+    }
+
+    async fn mediaserver_dir_tree(device: &DlnaDevice) -> RecVec<String, String> {
+    // fn mediaserver_dir_tree() -> RecVec<String, String> {
+        let mut node = RecVec {
+            // id: "Hello".to_string(),
+            id: device.id.to_string(),
+            value: device.name.to_string(),
+            // value: "World".to_string(), //device.id.to_string(),
+            children: Vec::new(),
+        };
+        let mut ms = MediaServerController::new(device.clone());
+        if let Ok(result) = ms.browse_directory(device.name.clone()).await {
+            node = Self::mediaserver_add_container(result, node);
+        }
+
+        node
+    }
+
+    fn mediaserver_add_container(container: MediaContainer, mut root: RecVec<String, String>) -> RecVec<String, String> {
+        let mut node = Self::mediaserver_item(container.id, container.name);
+        for child in container.childs {
+            node = Self::mediaserver_add_container(child, node);
+        }
+        /*let artists = container.items.iter().map(|artist_item| {
+            let artist_name = artist_item.artist.clone().unwrap();
+            let mut artist_node = Self::mediaserver_item(artist_item.id.clone(), artist_name.clone());
+            let albums: Vec<RecVec<String, String>> = container.items.iter()
+                .filter(|x| x.artist==Some(artist_name.clone()))
+                .map(|album_item| {
+                    let album_name = album_item.album.clone().unwrap();
+                    let mut album_node = Self::mediaserver_item(album_item.id.clone(), album_name.clone());
+                    let songs: Vec<RecVec<String, String>> = container.items.iter()
+                        .filter(|x| x.album==Some(album_name.clone()) && x.artist==Some(artist_name.clone()))
+                        .map(|track_item| {
+                            let element = format!("{} {}", track_item.track.clone(), track_item.title.clone());
+                            let song = Self::mediaserver_item(track_item.url.clone(), element);
+                            song
+                        }).collect();
+                    for song in songs {
+                        album_node.children.push(song);
+                    }
+                    album_node
+            }).collect();
+            for album in albums {
+                artist_node.children.push(album);
+            }
+            artist_node
+        });
+        for artist in artists {
+            node.children.push(artist);
+        } */
+        let mut songs = 0;
+        for item in container.items {
+            let album = item.album.unwrap_or("Unknown album".to_string());
+            let artist = item.artist.unwrap_or("Unknown artist".to_string());
```

Comment:
Consider using the constants from [`lib::library_db::track_db`](https://github.com/tramhao/termusic/blob/ac84456acaec365017dbb52435f5bcc5fb6a2273/lib/src/library_db/track_db.rs#L95) (i know they are in the library currently, but with #511 they would be moved, and they are also currently used elsewhere).